FirebaseCore.podspec

```diff
@@ -68,11 +73,35 @@ Firebase Core includes FIRApp and FIROptions which provide central configuration
       'FirebaseCore/Tests/Unit/**/*.[mh]',
       'SharedTestUtilities/FIROptionsMock.[mh]',
     ]
+
+    unit_tests.exclude_files = [
+      'FirebaseCore/Tests/Unit/FIRHeartbeatLoggerTest.m',
+    ]
+
     unit_tests.requires_app_host = true
     unit_tests.dependency 'OCMock'
     unit_tests.resources = 'FirebaseCore/Tests/Unit/Resources/GoogleService-Info.plist'
   end
 
+  # Using environment variable because of the dependency on the unpublished
+  # HeartbeatLoggingTestUtils.
+  if ENV['POD_LIB_LINT_ONLY'] && ENV['POD_LIB_LINT_ONLY'] == '1' then
+    s.test_spec 'heartbeat-unit' do |unit_tests|
+      unit_tests.scheme = { :code_coverage => true }
+      unit_tests.platforms = {
+        :ios => ios_deployment_target,
+        :osx => osx_deployment_target,
+        :tvos => tvos_deployment_target
+      }
+      unit_tests.source_files = [
+        'FirebaseCore/Tests/Unit/FIRHeartbeatLoggerTest.m',
+        # TODO: Is testing HeartbeatLogging on SPM enough?
```

Comment:
That would be fine - in which case this test_spec should be deleted. Do you understand why it doesn't work?

Reply:
Why what doesn't work– the mixed swift and objc unit tests? I'm not sure. I added the swift tests and then saw I had to change imports so I figured i'd hold off until we decided what we wanted

Reply:
So if this test_spec is deleted, I can wrap the "unit" test_spec in the env var check. Is OK for FirebaseCore's "unit" test spec to be gated by the env var?

Reply:
Yes. It's fine to use env vars in test specs.

Reply:
So if i set the env var locally and then use the `pod gen` tool to generate a workspace, the gated scheme does not work. Is this expected?

Reply:
It looks like it's set to intentionally fail:
![Screen Shot 2021-12-23 at 3 13 05 PM](https://user-images.githubusercontent.com/73870/147298657-a0ace9b8-aee6-45ab-ad69-d6935139d576.png)

Reply:
Hmm I cant even run that scheme.

Reply:
<img width="271" alt="Screen Shot 2021-12-23 at 6 21 13 PM" src="https://user-images.githubusercontent.com/36927374/147298998-ba886506-8f67-4777-8f95-a2402e8b4cdc.png">